plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/utils/AutoTooltip.java

```diff
@@ -0,0 +1,160 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.common.ui.utils;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.StyledText;
+import org.eclipse.swt.events.ControlEvent;
+import org.eclipse.swt.events.ControlListener;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+
+/**
+ * This class manages control's tooltip that should show text representation of the control 
+ * only when it is not fit into the space allocated to the control. If a custom tooltip 
+ * provided, the text representation of the control is added before it when needed.
+ * 
+ * Be aware that this class overrides any other tooltip set on the control.
+ *  
+ * @author Viacheslav Kabanovich
+ *
+ */
+public class AutoTooltip {
```

Comment:
IMHO AutoTooltip as a name is not very expressive to get what this class does. Hard to find a short and more expressive one though:
TooltipOnCroppedWidgetText, ... ?